bgpd/bgp_route.c

```diff
@@ -3190,6 +3191,18 @@ static void bgp_process_main_one(struct bgp *bgp, struct bgp_dest *dest,
 		bgp_unregister_for_label(dest);
 	}
 
+	if (safi == SAFI_MPLS_VPN && new_select && new_select->extra &&
+	    new_select->extra->bgp_orig)
+		new_bgp_orig = new_select->extra->bgp_orig;
+	if (new_bgp_orig && CHECK_FLAG(new_bgp_orig->vpn_policy[afi].flags,
```

Comment:
Why can't we combine two `IFs` into a single here?

Reply:
we can do, but I would need your confirmation.
From my perspective, the question is more: do we want to benefit from an intermediate new_bgp_orig variable ?
here is an output:
4 places that we could shortcut. 


```
if (safi == SAFI_MPLS_VPN && new_select && new_select->extra &&
	    new_select->extra->bgp_orig && CHECK_FLAG(new_select->extra->bgp_orig->vpn_policy[afi].flags,
BGP_VPN_POLICY_TOVPN_LABEL_TO_SEND)) {
		zclient_send_vrf_label(zclient, new_select->extra->bgp_orig->vrf_id, afi,
				       new_select->extra->bgp_orig->vpn_policy[afi]
					       .tovpn_zebra_vrf_label_to_send,
				       ZEBRA_LSP_BGP);
		UNSET_FLAG(new_select->extra->bgp_orig->vpn_policy[afi].flags,
			   BGP_VPN_POLICY_TOVPN_LABEL_TO_SEND);
	}
```

if there is any agreement, I will incline.
but please tell me.

Reply:
do `new_bgp_orig = new_select->extra->bgp_orig;`

Reply:
please combine the if